code/obj/item/device/radios/intercoms.dm

```diff
@@ -94,6 +94,48 @@ TYPEINFO(/obj/item/device/radio/intercom)
 	var/maptext = generateMapText(msg, textLoc, style = "color:[color];", alpha = 255)
 	target.show_message(type = 2, just_maptext = TRUE, assoc_maptext = maptext)
 
+/obj/item/device/radio/intercom/receive_silicon_hotkey(var/mob/user)
+	..()
+
+	if (!isAI(user))
+		return
+
+	if (!isAIeye(user))
+		boutput("Deploy to an AI Eye first to override intercoms.")
+
+	if(user.client.check_key(KEY_BOLT))
+		if (src.locked_frequency)
+			boutput(user, "<span class='alert'>You can't override an intercom with a locked frequency!</span")
+			return
+
+		var/original_src_frequency = src.frequency
+		var/original_src_broadcasting = src.broadcasting
+		var/original_src_listening = src.listening
+		var/text_colour = "#CC3FCC"
+
+		// fake it till you make it
+		var/message = "<span class='radio [src.chat_class]' style='color:[src.device_color || text_colour]'>[radio_icon(src)]\
+		<span class='name'>[src]</span> <span class='message'>alerts, \"AI override engaged!\"</span></span>"
+		var/maptext = make_chat_maptext(src, "AI override engaged!", "color:[text_colour]")
+
+		src.speech_bubble(image('icons/mob/mob.dmi', "ai"))
+		for (var/i in hearers(7, src.loc))
+			if (ismob(i))
```

Comment:
do the filtering in the loop `for (var/mob/M in hearers(...))` etc
I believe it should be more or less perf-neutral currently, but will be a speedup once we upgrade to BYOND 515